### PR TITLE
default user handling

### PR DESCRIPTION
**Pull request by laliconfigcat**
### Describe the purpose of your pull request

Adding the default user functionality to the SDK.

### Requirement checklist (only if applicable)

- [x] I have covered the applied changes with automated tests.
- [x] I have executed the full automated test set against my changes.
- [x] I have validated my changes against all supported platform versions.
- [ ] I have read and accepted the [contribution agreement](https://github.com/configcat/legal/blob/main/contribution-agreement.md).

### CODE REVIEW

File: test/DefaultUserTests.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+describe("DefaultUser", () => {
+
+  it("Default user set works", async () => {
+    const redEyeColorUser = { identifier: 'redIdentifier', custom: { 'eyeColor': 'red' } };
+    const blueEyeColorUser = { identifier: 'blueIdentifier', custom: { 'eyeColor': 'blue' } };
+
+    let configCatKernel: FakeConfigCatKernel = { configFetcher: new FakeConfigFetcherWithRules(), sdkType: "common", sdkVersion: "1.0.0" };
+    var client: IConfigCatClient = configcatClient.createClientWithAutoPoll("APIKEY", configCatKernel, {logger: configcatClient.createConsoleLogger(configcatClient.LogLevel.Debug)});
+
+    // Default
```
Comment (sigewuzhere) [inline test/DefaultUserTests.ts]: Please be a bit more specific about what these tests are about

**Review by sigewuzhere — CHANGES_REQUESTED** [review]

File: test/DefaultUserTests.ts (hunk 1 above)
Comment (laliconfigcat) [reply test/DefaultUserTests.ts]: What about now?

**Review by sigewuzhere — APPROVED** [review]

File: src/ConfigCatClient.ts [hunk 2]
```diff
@@ -55,12 +55,20 @@ export interface IConfigCatClient {
 
     /** Returns the values of all feature flags or settings */
     getAllValuesAsync(user?: User): Promise<SettingKeyValue[]>;
+
+    /** Sets the default user for feature flag evaluations. 
+     * In case the getValue function isn't called with a UserObject, this default user will be used instead. */
+    setDefaultUser(defaultUser?: User): void;
```
Comment (endret) [inline src/ConfigCatClient.ts]: nullable parameter and `clearDefaultUser()` look redundant to me, can you explain why it is necessary (nullable parameter)?

Comment (laliconfigcat) [reply src/ConfigCatClient.ts]: I think it gives more flexibility in using this feature.

Comment (sigewuzhere) [reply src/ConfigCatClient.ts]: Yes, plus a nullable parameter needs to be explained, `clearDefaultUser()` is self-explanatory.

**Review by endret — APPROVED** [review]

File: src/ConfigCatClient.ts (hunk 2 above)
Comment (laliconfigcat) [reply src/ConfigCatClient.ts]: I updated the setDefaultUser to non nullable

**Review by sigewuzhere — APPROVED** [review]

File: src/ConfigCatClient.ts (hunk 2 above)
Comment (endret) [reply src/ConfigCatClient.ts]: It would be nice to update the interface too.

Comment (laliconfigcat) [reply src/ConfigCatClient.ts]: done

**Review by endret — APPROVED** [review]

**Pull request merged**